hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/client/RatisReplicationConfig.java

```diff
@@ -35,6 +35,16 @@ public RatisReplicationConfig(ReplicationFactor replicationFactor) {
     this.replicationFactor = replicationFactor;
   }
 
+  public RatisReplicationConfig(String factorString) {
+    ReplicationFactor factor = null;
+    try {
+      factor = ReplicationFactor.valueOf(Integer.parseInt(factorString));
+    } catch (NumberFormatException ex) {
+      factor = ReplicationFactor.valueOf(factorString);
```

Comment:
Should we throw an exception explicitly here?

Reply:
In general the expected values are `ONE` and `THREE` but this code adds support to parse `ozone.replication` which can be 1 and 3 today. Long-term it will enable to use `3:2` in the `ozone.replication` (but we also need to configure EC as default `replicationType`)

Reply:
> Should we throw an exception explicitly here?

Currently, `IllegalArgumentException` is thrown with the message:

```
Exception in thread "main" java.lang.IllegalArgumentException: No enum constant org.apache.hadoop.hdds.protocol.proto.HddsProtos.ReplicationFactor.asd
```
I think the type (`IllegalArgumentException`) is fine, but I modified the error message with throwing explicit exception:

```
Exception in thread "main" java.lang.IllegalArgumentException: Invalid RatisReplicationFactor 'xxx'. Please use ONE or THREE
```

Reply:
Thanks for the update. At first, I thought ReplicationFactor is only for legacy configs and won't be used for EC schemas.